apps/metadata-editor/src/app/edit/components/publish-button/publish-button.component.ts

```diff
@@ -42,9 +48,48 @@ export class PublishButtonComponent {
     })
   )
 
-  constructor(private facade: EditorFacade) {}
+  record$ = this.facade.record$
+  activeUser$: Observable<UserModel>
+
+  constructor(
+    private facade: EditorFacade,
+    private recordsApiService: RecordsApiService,
+    private platformService: PlatformServiceInterface,
+    private groupApiService: GroupsApiService
+  ) {
+    this.activeUser$ = this.platformService.getMe()
+  }
 
   saveRecord() {
     this.facade.saveRecord()
+    this.facade.saveSuccess$
+      .pipe(
+        take(1),
+        switchMap(() =>
+          combineLatest([this.activeUser$, this.record$]).pipe(take(1))
+        )
+      )
+      .subscribe(([userId, record]) => {
+        this.groupApiService.getGroups().subscribe((groups) => {
+          const groupId = groups.find(
+            (grp) => grp.name === record.ownerOrganization.name
+          )
+          // setting to 0 by default because the API expects a number and not null & most records don't have a group
+          this.recordsApiService
+            .setRecordOwnership(
+              record.uniqueIdentifier,
+              groupId ? groupId.id : 0,
+              Number(userId.id)
+            )
+            .subscribe(
+              () => {
+                console.log('Ownership set successfully')
+              },
+              (error) => {
+                console.log('Error setting ownership :', error)
+              }
```

Comment:
I get it for dev purpose, but I don't think we should pollute the console log with this.